src/mem.c

```diff
@@ -89,12 +89,19 @@ kore_realloc(void *ptr, size_t len)
 void *
 kore_calloc(size_t memb, size_t len)
 {
-	if (memb == 0 || len == 0)
+	void *result;
+	const size_t nbytes = memb * len;
+
+	if (nbytes == 0)
 		fatal("kore_calloc(): zero size");
-	if (SIZE_MAX / memb < len)
```

Comment:
Hi Joris - 

Ah, I’ve looked again and now I understand your intent with potential size_t overflow of memb * len.  Thanks for pointing that out.

More generally I’m curious about why you’re reimplementing malloc, calloc, realloc in this way?  The realloc in particular is going to be slower than the glibc implementation, and it gets used on the socket receive path.

Best wishes,
Stig


> On 30 Dec 2015, at 23:44, Joris Vink <notifications@github.com> wrote:
> 
> Why did you remove the overflow check in kore_calloc() though? It is now susceptible to an integer overflow. The check you introduced is just plain wrong.
> 
>